server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -85,14 +86,17 @@ public QueryResource(
       QuerySegmentWalker texasRanger,
       ServiceEmitter emitter,
       RequestLogger requestLogger,
-      QueryManager queryManager
+      QueryManager queryManager,
+      InjectableValues guiceInjectableValues
   )
   {
     this.config = config;
     this.jsonMapper = jsonMapper.copy();
+    this.jsonMapper.setInjectableValues(guiceInjectableValues); // Workaround for https://github.com/FasterXML/jackson-databind/issues/696
```

Comment:
From what I can tell, nothing is actually closing the OutputStream, which makes me think that it is the Jersey framework that is taking on that responsibility.  

If that is the case, then I really think that it is the Jersey framework's job to make sure that calling `close()` on the `OutputStream` doesn't screw with things.  But if they haven't done that, it should be easy enough for us to ensure that `close()` doesn't get called.